antiope-sqs/src/Antiope/SQS.hs

```diff
@@ -61,16 +62,19 @@ ackMessages :: (MonadAWS m, HasReceiptHandle msg)
   -> [msg]
   -> m (Either SQSError ())
 ackMessages (QueueUrl queueUrl) msgs = do
-  let receipts = msgs ^.. each . to getReceiptHandle & catMaybes
-  -- each dmbr needs an ID. just use the list index.
-  let dmbres = (\(r, i) -> deleteMessageBatchRequestEntry (pack (show i)) r) <$> zip (coerce receipts) ([0..] :: [Int])
-  resp <- AWS.send $ deleteMessageBatch queueUrl & dmbEntries .~ dmbres
-  -- only acceptable if no errors.
-  if resp ^. dmbrsResponseStatus == 200
-    then case resp ^. dmbrsFailed of
-        [] -> return $ Right ()
-        _  -> return $ Left DeleteMessageBatchError
-    else return $ Left DeleteMessageBatchError
+  let receipts' = msgs ^.. each . to getReceiptHandle & catMaybes
+  results <- forM (chunksOf 10 receipts') $ \receipts -> do
```

Comment:
nit: would be good to make the `10` as something like `maxBatchSize` for more explicit way.

Reply:
Sure